pennylane/plugins/default_qubit.py

```diff
@@ -325,11 +328,18 @@ def reset(self):
         self._state = self._create_basis_state(0)
         self._pre_rotated_state = self._state
 
-    def analytic_probability(self, wires=None):
-        """Return the (marginal) analytic probability of each computational basis state."""
+    def analytic_probability(self, subsystems=None):
+        """Return the (marginal) analytic probability of each computational basis state.
+
+        Args:
+            subsystems (Sequence[int]): indices of the target wires
+
+        Returns:
+            array: vector of marginal probabilities
+        """
```

Comment:
:+1:

Reply:
Oh actually this is inherited from QubitDevice, so doesn't need a docstring?

Reply:
This is part of the QubitDevice API, so I'm not sure should change, actually? Since some devices might want qubit label rather than index? I'm not 100% sure.

Reply:
That should be no problem. I could make all the ``qubit_device`` functions take ``wires`` in the QubitDevice API, and in this case just translate inside the function. Would that be more flexible?

Reply:
I think so, yeah!

Reply:
Sounds good, done.

Reply:
I also rewrote the ``device`` in a similar manner, so it passes the wires to the devices, which then need to extract the ``registers`` or indices of the wires on the register...

(I hoped previously we could do this after the ``Device`` class refactor, but better to do it now!)

The internal devices are refactored in this PR, but we need to update all plugins when merging this PR. I'll start making PRs in the plugin repos.